File: Signiture-Backend/delivery/db/src/main/java/pcy/study/db/storemenu/StoreMenu.java
package pcy.study.db.storemenu;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import pcy.study.db.BaseTimeEntity;
import pcy.study.db.storemenu.enums.StoreMenuStatus;

import java.math.BigDecimal;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class StoreMenu extends BaseTimeEntity {

    @Column(nullable = false)
    private Long storeId;

    @Column(length = 100, nullable = false)
    private String name;

    @Column(precision = 11, scale = 4, nullable = false)
    private BigDecimal amount;

    @Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @Enumerated(EnumType.STRING)
    private StoreMenuStatus status;

    @Column(length = 200, nullable = false)
    private String thumbnailUrl;

    private int likeCount;
    private int sequence;

    @Builder
    private StoreMenu(Long storeId, String name, BigDecimal amount, String thumbnailUrl) {
        this.storeId = storeId;
        this.name = name;
        this.amount = amount;
        this.thumbnailUrl = thumbnailUrl;
        this.status = StoreMenuStatus.REGISTERED;
    }
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/storemenu/StoreMenu.kt
package pcy.study.db.storemenu

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import pcy.study.db.store.Store
import pcy.study.db.storemenu.enums.StoreMenuStatus
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
data class StoreMenu @JvmOverloads constructor(

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null,

    @field:ManyToOne
    @field:JoinColumn(name = "store_id", nullable = false)
    var store: Store,

    @field:Column(length = 100, nullable = false)
    var name: String,

    @field:Column(precision = 11, scale = 4, nullable = false)
    var amount: BigDecimal,

    @field:Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @field:Enumerated(EnumType.STRING)
    var status: StoreMenuStatus? = StoreMenuStatus.REGISTERED,

    @field:Column(length = 200, nullable = false)
    var thumbnailUrl: String,

    var likeCount: Int? = 0,
    var sequence: Int? = 0,

    @field:CreationTimestamp
    @field:Column(updatable = false)
    private var createdAt: LocalDateTime? = null,

    @field:UpdateTimestamp
    private var updatedAt: LocalDateTime? = null,
)